kafka-connect-aws-s3/README-source.md

```diff
@@ -0,0 +1,183 @@
+    connector.class=io.lenses.streamreactor.connect.aws.s3.source.S3SourceConnector
+    tasks.max=1
+    connect.s3.kcql=insert into $TOPIC_NAME select * from $BUCKET_NAME:$PREFIX_NAME STOREAS `parquet`
+    aws.region=eu-west-1
+    aws.secret.key=SECRET_KEY
+    aws.access.key=ACCESS_KEY
+    aws.auth.mode=Credentials
+    value.converter=io.confluent.connect.avro.AvroConverter
+    value.converter.schema.registry.url=http://localhost:8089
+
+You should replace $BUCKET_NAME, $PREFIX_NAME and $TOPIC_NAME with the names of the bucket, desired prefix and topic.
+
+ACCESS_KEY and SECRET_KEY are credentials generated within AWS IAM and must be set and configured with permissions to write to the desired S3 bucket.
+
+Please read below for a detailed explanation of these and other options, including the meaning of WITH_FLUSH_COUNT and its alternatives.
+
+
+### Source Format configuration
+
+Format configuration is provided by kcql.
+
+The options for json, avro and parquet will look like the below:
+
+    connect.s3.kcql=insert into $TOPIC_NAME select * from $BUCKET_NAME:$PREFIX_NAME STOREAS `JSON`
+
+The options for the formats are case-insensitive, but they are presented here in the most readable form.
+
+#### JSON Input Format Configuration
+
+Using JSON as an input format allows you to read in the JSON-written sink files, line by line.
+
+    STOREAS `JSON`
+
+Please note: The JSON is not parsed by the S3 Source connector.  It is recommended to configure the JsonConverter if you wish the Json to be parsed on entry into Kafka.
+
+    value.converter=org.apache.kafka.connect.json.JsonConverter
+    value.converter.schemas.enable=false
+
+#### Avro Input Format Configuration
+
+Using Avro as the input format allows you to read the Avro-stored messages on S3 back into Kafka's native format.
+
+    STOREAS `Avro`
+    
+    
+It may also be necessary to configure the message converter:
+
+    value.converter=io.confluent.connect.avro.AvroConverter
+    value.converter.schema.registry.url=http://localhost:8089 
+    
+#### Parquet Input Format Configuration
+
+Using Parquet as the input format allows you to read parquet files stored on S3, importing the Avro schemas and values.
+
+    STOREAS `Parquet`
+    
+It may also be necessary to configure the message converter:
+
+    value.converter=io.confluent.connect.avro.AvroConverter
+    value.converter.schema.registry.url=http://localhost:8089 
+
+
+#### Text Input Format Configuration
+
+If the source files on S3 consist of files containing lines of text, then using the text input format may be desired. 
+
+    STOREAS `Text`
+    
+It may be required to use the additional configuration options for this connector to ensure that the value is presented as a String.
+
+    value.converter=org.apache.kafka.connect.storage.StringConverter
+    
+
+#### CSV Input Format Configuration
+
+This reads CSV files written to S3 into an Avro struct to be written back to the Kafka source.
```

Comment:
I don't think we can ever do that unless each field is market as String. Is this happening?

Reply:
The opencsv CSV reader provides us with String representations of all of the fields that were written out to CSV.

The converting to struct is an approach suggested by @andrewstevenson so it would be good to get him involved in the discussion.

Reply:
CSV parsing has now been removed.